optuna/visualization/matplotlib/_parallel_coordinate.py

```diff
@@ -1,11 +1,130 @@
 
 
 @experimental("2.2.0")
-def plot_parallel_coordinate() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_parallel_coordinate(study: Study, params: Optional[List[str]] = None) -> "Axes":
+    """Plot the high-dimentional parameter relationships in a study with Matplotlib.
+
+    .. seealso::
+        Please refer to :func:`optuna.visualization.plot_parallel_coordinate` for an example.
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+        params:
+            Parameter list to visualize. The default is all parameters.
+
+    Returns:
+        A :class:`matplotlib.axes.Axes` object.
+    """
+
+    _imports.check()
+    return _get_parallel_coordinate_plot(study, params)
+
+
+def _get_parallel_coordinate_plot(study: Study, params: Optional[List[str]] = None) -> "Axes":
+
+    # Set up the graph style.
+    fig, ax = plt.subplots()
+    cmap = plt.get_cmap("Blues")
+    ax.set_title("Parallel Coordinate Plot")
+    ax.spines["top"].set_visible(False)
+    ax.spines["bottom"].set_visible(False)
+
+    # Prepare data for plotting.
+    trials = [trial for trial in study.trials if trial.state == TrialState.COMPLETE]
+
+    if len(trials) == 0:
+        _logger.warning("Your study does not have any completed trials.")
+        return ax
+
+    all_params = {p_name for t in trials for p_name in t.params.keys()}
+    if params is not None:
+        for input_p_name in params:
+            if input_p_name not in all_params:
+                raise ValueError("Parameter {} does not exist in your study.".format(input_p_name))
+        all_params = set(params)
+    sorted_params = sorted(list(all_params))
+
+    obj_org = [t.value for t in trials]
+    obj_min = min(obj_org)
+    obj_max = max(obj_org)
+    obj_w = obj_max - obj_min  # type: ignore
+    dims_obj_base = [[o] for o in obj_org]
+
+    cat_param_names = []
+    cat_param_values = []
+    cat_param_ticks = []
+    param_values = []
+    var_names = ["Objective Value"]
+    for p_name in sorted_params:
+        values = [t.params[p_name] if p_name in t.params else np.nan for t in trials]
+        try:
+            tuple(map(float, values))
+        except (TypeError, ValueError):
+            vocab = defaultdict(lambda: len(vocab))  # type: DefaultDict[str, int]
+            values = [vocab[v] for v in values]
+            cat_param_names.append(p_name)
+            vocab_item_sorted = sorted(vocab.items(), key=lambda x: x[1])
+            cat_param_values.append([v[0] for v in vocab_item_sorted])
+            cat_param_ticks.append([v[1] for v in vocab_item_sorted])
+
+        p_min = min(values)
+        p_max = max(values)
+        p_w = p_max - p_min
+        for i, v in enumerate(values):
+            dims_obj_base[i].append((v - p_min) / p_w * obj_w + obj_min)
+
+        var_names.append(p_name if len(p_name) < 20 else "{}...".format(p_name[:17]))
```

Comment:
[question] why is the name truncated at 17?

Reply:
This one is based on the Plotly backend.
https://github.com/optuna/optuna/blob/f6ebb4c640164b5bece2357cfd933610428c71b5/optuna/visualization/_parallel_coordinate.py#L105

Reply:
Ah, got it. Thanks 😃